internal/tools/argocd/argocd.go

```diff
@@ -0,0 +1,83 @@
+package argocd
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+type ArgocdClusterLabels struct {
+	ClusterTypes string `yaml:"cluster-type"`
+	ClusterID    string `yaml:"cluster-id"`
+}
+
+type ArgocdClusterRegisterParameters struct {
+	Name      string              `yaml:"name"`
+	Type      string              `yaml:"type"`
+	Labels    ArgocdClusterLabels `yaml:"labels"`
+	APIServer string              `yaml:"api_server"`
+	CertData  string              `yaml:"certData"`
+	CaData    string              `yaml:"caData"`
+	KeyData   string              `yaml:"keyData"`
+}
+
+type Argock8sRegister struct {
+	Clusters []ArgocdClusterRegisterParameters `yaml:"clusters"`
+}
+
+// ReadArgoK8sRegistrationFile take a argocd cluster file and load it into Argock8sRegister struct
+func (a *Argock8sRegister) ReadArgoK8sRegistrationFile(clusterFile []byte) (*Argock8sRegister, error) {
+
+	if err := yaml.Unmarshal(clusterFile, a); err != nil {
+		return nil, errors.Wrap(err, "Error unmarshaling Argo k8s cluster YAML definition")
+	}
+	return a, nil
+}
```

Comment:
I think we just want this to be a normal function. Something like this:
```suggestion
// ReadArgoK8sRegistrationFile take a argocd cluster file and load it into Argock8sRegister struct
func ReadArgoK8sRegistrationFile(clusterFile []byte) (*Argock8sRegister, error) {
        var a &argocd.Argock8sRegister
	if err := yaml.Unmarshal(clusterFile, a); err != nil {
		return nil, errors.Wrap(err, "Error unmarshaling Argo k8s cluster YAML definition")
	}
	return a, nil
}
```